src/main/java/org/myrobotlab/service/InMoov.java

```diff
@@ -154,8 +158,20 @@ public InMoov(String n) {
   // ---------------------------------------------------------------
 
   public void attach(Attachable attachable) {
+    //opencv
     if (attachable instanceof OpenCV) {
       opencv = (OpenCV) attachable;
+      subscribe(opencv.getName(), "publishClassification");
+      //SpeechRecognizer
+    } else if (attachable instanceof SpeechRecognizer) {
+
+      ear = (SpeechRecognizer) attachable;
+    } else if (attachable instanceof ProgramAB) {
+      chatBot = (ProgramAB) attachable;
+    }
+    if (mouth != null && ear != null) {
+      //TODO attach...
+      ear.addMouth(mouth);
```

Comment:
normally in python you would be doing this.  I don't think this is the right place for it.  This means that any time any service is attached after the ear and mouth have been attached,  this ear.addMouth method is going to get called.

Reply:
If we can offer some shortcuts ... I prefer not to remove the method for the moment because it was already launched while startMouth(). I reviewed the code so that it would be executed only 1 time. it would be great to have a global x.isAttached(y) also ..